socorro/cron/jobs/ftpscraper.py

```diff
@@ -335,6 +345,16 @@ def download(self, url):
         assert response.status_code == 200, response.status_code
         return response.content
 
+    def skip_json_file(self, json_url):
+        basename = os.path.basename(json_url)
+        for rule in self.config.json_files_to_ignore:
+            # If the rule was, for example, `*.foo.bar` then,
+            # the regex becomes `.\.foo\.bar`.
+            regex_string = re.escape(rule).replace('\\*', '.')
```

Comment:
I'm pretty sure this isn't right. This replaces a `*` with a regex `.` which matches a character of length 1.

I think what you're trying to do here is convert from a file-globbing style syntax where `*` is a wildcard for 0 or more characters. If that's the case, then you probably want to convert `*` to `.*`.

Further, I claim you also want to add anchors to the regexp and use `.match` instead of `.findall`. So then you're converting something like `*.foo.bar` to `^.*\.foo\.bar$`. This prevents substring matching so `*.foo` doesn't accidentally match `test.foo.baz` which I don't think you want.

Does that sound right? If so, then I think this line would be:

```
regex_string = '^' + re.escape(rule).replace(r'\*', '.*') + '$'
```

I'm a little iffy on the placement of that re.escape. I'm pretty sure that creates weird edge cases with escaping. I'm not sure if it matters.

Regardless, I suggest another approach which has the nicety of also supporting `?` and `[]` notation: use the fnmatch module. https://docs.python.org/2/library/fnmatch.html